src/validation/ValidationContext.js

```diff
@@ -42,11 +42,44 @@ type VariableUsage = {|
  * allowing access to commonly useful contextual information from within a
  * validation rule.
  */
-export default class ValidationContext {
-  _schema: GraphQLSchema;
+export class BaseValidationContext {
```

Comment:
I like this split, as there's so many rules that don't need a schema at all.

However, I am beginning to think there are three pieces:
- a pure "ErrorReporter". Has no AST knowledge.
- an AST-only context. This is `SDLValidationContext` as it stands. Everything that is a pure `visitor` can use this context. It may need two documents, Schema and Executable ASTs.
- a Schema-aware context. Much like `visitor` can take in a `visitWithTypeInfo`, this would hold a `ASTValidationContext`, and delegate to that one for the ast functionality.

Reply:
Addressed in #1446